test/test_base.py

```diff
@@ -17,45 +19,63 @@ def __init__(self):
         self.network_name = self.tmpdir.name
         self.logfile = None
         self.server = None
+        self.network = False
 
         atexit.register(self.cleanup)
 
         print(f"\nWarnet test base started")
 
-
-    def cleanup(self, signum = None, frame = None):
-        if self.server is None:
-            return
-
+    def _cleanup_docker(self):
         try:
-            print("\nStopping network")
-            self.warcli("network down")
-
-            self.wait_for_all_tanks_status(target="none", timeout=60, interval=1)
-
-            print("\nStopping server")
-            self.warcli("stop", False)
-        except Exception as e:
-            # Remove the temporary docker network when we quit.
-            # If the warnet server exited prematurely then docker-compose down
-            # likely did not succeed or was never executed.
-            print(f"Error stopping server: {e}")
             print("Attempting to cleanup docker network")
             wn = Warnet.from_network(self.network_name)
             wn.docker_compose_down()
+        except FileNotFoundError:
+            pass
+        except Exception:
+            pass
+
+    def _cleanup_network(self):
+        if self.network:
+            try:
+                print("\nStopping network")
+                self.warcli("network down")
+
+                self.wait_for_all_tanks_status(target="none", timeout=60, interval=1)
+
+                print("\nStopping server")
+                self.warcli("stop", False)
+            except Exception as e:
+                # Remove the temporary docker network when we quit.
+                # If the warnet server exited prematurely then docker-compose down
+                # likely did not succeed or was never executed.
+                print(f"Error stopping server: {e}")
+                self._cleanup_docker()
+
+    def _cleanup_server(self):
+        # kill the server process by its PID if it still survives
+        if self.server:
+            print(f"killing server with pid {self.server.pid}")
+            os.killpg(os.getpgid(self.server.pid), signal.SIGTERM)
```

Comment:
this looks to me like the most useful part of this PR, I dont really understand the need for the other changes. Why couldn't this emergency termination just be part of the original `cleanup()` in the exception handler (i.e. if `warcli stop` fails?)